c.c

```diff
@@ -1388,10 +1388,16 @@ static void skipToMatch (const char *const pair)
 	{
 		if (CollectingSignature)
 			vStringPut (Signature, c);
-		
 
 		if (c == begin)
 		{
+			char s = cppGetc();
+			if (s == '<' && s == c) {
+				// templates can still contain "<<" ot "<<="
+				break;
```

Comment:
I can't really understand what you mean, this part of the patch looks for "<<" pattern inside a template, so going on parsing while ignoring the shift should be safe. 
The other part takes care of not failing completely if we are unable to find the closing '>' 

Whit this patch the snippet 
```
template <int P> class c {};
c< 8 > bVar;
c< 1<<8 > aVar;
c< 1>>8 > cVar;
```
produces
```
bVar	/home/martino/arduino-builder/src/arduino.cc/builder/test/sketch11/sketch_fastleds.ino	/^c< 8 > bVar;$/;"	kind:variable	line:30
aVar	/home/martino/arduino-builder/src/arduino.cc/builder/test/sketch11/sketch_fastleds.ino	/^c< 1<<8 > aVar;$/;"	kind:variable	line:31
cVar	/home/martino/arduino-builder/src/arduino.cc/builder/test/sketch11/sketch_fastleds.ino	/^c< 1>>8 > cVar;$/;"	kind:variable	line:32
```

Should be ok, right? :smile: